tests/system/apiserver-reconnection.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash -ex
+
+# Utilities.
+function get_container_ip {
+    docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' $1
+}
+
+function create_namespace {
+    name=$1
+    curl -k -H "Content-Type: application/yaml" -XPOST --data-binary @- https://172.17.0.3:6443/api/v1/namespaces <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ${name}
+EOF
+    sleep 1
+}
+
+ETCD_IP=`get_container_ip st-etcd`
+K8S_IP=`get_container_ip st-apiserver`
+
+# Run policy controller.
+docker rm -f calico-policy-controller || true
+sleep 2
+docker run --detach --name=calico-policy-controller \
+       -e K8S_API=https://${K8S_IP}:6443 \
+       -e K8S_INSECURE_SKIP_TLS_VERIFY=true \
+       -e ETCD_ENDPOINTS=http://${ETCD_IP}:2379 \
+       calico/kube-policy-controller
+sleep 2
+
+# Create a namespace.
+NS_NAME=chocolate
+create_namespace ${NS_NAME}
+
+# Check for that namespace in etcd.
+docker exec st-etcd etcdctl ls --recursive /calico | grep ${NS_NAME}
+
+for n in `seq 0 9`; do
```

Comment:
Do we really need to do this 9 times?  Is this just because the test itself doesn't catch problems every time?

I'd like to include ST as part of per-commit testing, but as it is this test will take half an hour on its own, which isn't really feasible.